Framework/Geometry/src/Objects/ShapeFactory.cpp

```diff
@@ -963,6 +1145,43 @@ Hexahedron ShapeFactory::parseHexahedron(Poco::XML::Element *pElem) {
   hex.rbb = parsePosition(pElem_rbb);
   hex.rbt = parsePosition(pElem_rbt);
 
+  // Rotate the points by the rotate, rotate-all and goniometer tags
+  if (pElem_rot) {
+    // Apply manual rotation supplied to rotate tag
+    std::vector<double> rotateAngles = DegreesToRadians(parsePosition(pElem_rot));
+    const std::vector<double> rotateMatrix = generateMatrix(rotateAngles[0], rotateAngles[1], rotateAngles[2]);
+    hex.lfb.rotate(rotateMatrix);
+    hex.lft.rotate(rotateMatrix);
+    hex.lbb.rotate(rotateMatrix);
+    hex.lbt.rotate(rotateMatrix);
+    hex.rfb.rotate(rotateMatrix);
+    hex.rft.rotate(rotateMatrix);
+    hex.rbb.rotate(rotateMatrix);
+    hex.rbt.rotate(rotateMatrix);
+  }
+  if (m_rotateAllMatrix.size() > 0) {
+    // Apply automatic rotation due to the rotate-all tag
+    hex.lfb.rotate(m_rotateAllMatrix);
+    hex.lft.rotate(m_rotateAllMatrix);
+    hex.lbb.rotate(m_rotateAllMatrix);
+    hex.lbt.rotate(m_rotateAllMatrix);
+    hex.rfb.rotate(m_rotateAllMatrix);
+    hex.rft.rotate(m_rotateAllMatrix);
+    hex.rbb.rotate(m_rotateAllMatrix);
+    hex.rbt.rotate(m_rotateAllMatrix);
+  }
+  if (m_gonioRotateMatrix.size() > 0) {
+    // Apply automatic rotation due to the goniometer that should NOT be manually defined by the user
+    hex.lfb.rotate(m_gonioRotateMatrix);
+    hex.lft.rotate(m_gonioRotateMatrix);
+    hex.lbb.rotate(m_gonioRotateMatrix);
+    hex.lbt.rotate(m_gonioRotateMatrix);
+    hex.rfb.rotate(m_gonioRotateMatrix);
+    hex.rft.rotate(m_gonioRotateMatrix);
+    hex.rbb.rotate(m_gonioRotateMatrix);
+    hex.rbt.rotate(m_gonioRotateMatrix);
+  }
+
```

Comment:
This is quite repetitive. I think it's worth adding a rotate method to the Hexahedron struct that rotates all 8 fields and then you can just call that method here. Although it's not essential, it's probably good "style" to change the struct to a class also